pkg/crd/desc_visitor.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crd
+
+import (
+	"strings"
+
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+// TruncateDescription truncates the description of fields in given schema if it
+// exceeds maxLen.
+// It tries to chop off the description at the closest sentence boundary.
+func TruncateDescription(schema *apiext.JSONSchemaProps, maxLen int) *apiext.JSONSchemaProps {
+	outSchema := schema.DeepCopy()
```

Comment:
Do we really need to deepcopy it? The schema may be huge.

Reply:
eh.  It's probably fine.  It's not like we're doing this at scale.  Although, I'd just move this out of the call, and make `TruncateDescription` not return a value, and then the caller can choose

Reply:
refactored it as suggested by @DirectXMan12